### PR TITLE
Add e2e tests for FA2

### PR DESCRIPTION
**Pull request by erman-gurses**
Initial tests of FA2 for CPU backend.

### CODE REVIEW

**Review by hanhanW — APPROVED** [review]

File: tests/e2e/linalg_ext_ops/attention.mlir [hunk 1]
```diff
@@ -12,3 +12,101 @@ func.func @attention() {
   ) : tensor<1x4x4xf32>
   return
 }
+
+func.func @attention1() {
```
Comment (hanhanW) [inline tests/e2e/linalg_ext_ops/attention.mlir]: I don't have a good suggestion, but I wonder if we can make function names more meaningful?

Comment (erman-gurses) [reply tests/e2e/linalg_ext_ops/attention.mlir]: Updated, thanks!

File: tests/e2e/linalg_ext_ops/attention.mlir [hunk 2]
```diff
@@ -12,3 +12,101 @@ func.func @attention() {
...
+  check.expect_almost_eq_const(
+      %1,
+      dense<[[[0.5530, 0.6530, 0.7530, 0.8530],
+              [0.6328, 0.7328, 0.8328, 0.9328],
+              [0.7011, 0.8011, 0.9011, 1.0011]]]> : tensor<1x3x4xf32>
+  ) : tensor<1x3x4xf32>
+  return
+}
+
+func.func @attention1x4x4() {
```
Comment (hanhanW) [inline tests/e2e/linalg_ext_ops/attention.mlir:25]: Perhaps we can retire the first test case, since this covers the needs? What do you think?

Comment (hanhanW) [reply tests/e2e/linalg_ext_ops/attention.mlir:25]: By the first test case, I meant `func.func @attention()`

**Review by hanhanW — APPROVED** [review]

File: tests/e2e/linalg_ext_ops/attention.mlir (hunk 2 above)
Comment (erman-gurses) [reply tests/e2e/linalg_ext_ops/attention.mlir:25]: I agree with you. The names will make more sense, then.

**Comment by Groverkss** [comment]
I'm not sure how realistic these tests are. Attention almost always has head_dim 64 or 128. Seq length is also at least 512.

A lot of pipelines should just use a dumb pipeline for these shapes, so it doesn't really test the pipelines we want to test.

**Comment by erman-gurses** [comment]
> I'm not sure how realistic these tests are. Attention almost always has head_dim 64 or 128. Seq length is also at least 512.

More tests will be added using Test-Generator. Currently, I am working on it.

**Comment by MaheshRavishankar** [comment]
> I'm not sure how realistic these tests are. Attention almost always has head_dim 64 or 128. Seq length is also at least 512.
> 
> A lot of pipelines should just use a dumb pipeline for these shapes, so it doesn't really test the pipelines we want to test.

I think its more about testing attention op on IREE as a whole. Which pipeline is used is an implementation detail of the backend. For IREE as a whole attention needs to be implemented and tested.

**Pull request merged**